### PR TITLE
Remove unneeded `@trusted` annotations

### PR DESCRIPTION
**Pull request by dkorpel**
Most backend modules have `@safe:` on top with `@trusted` bulk added to individual functions. Lots of those are no longer needed as the calls to functions that made them `@system` are now also `@trusted`.

### CODE REVIEW

File: compiler/src/dmd/backend/dtype.d [hunk 1]
```diff
@@ -1192,7 +1192,6 @@ uint param_t_length(scope param_t* p)
  * Input:
  *      ptali   initial template-argument-list
  */
-
 @trusted
```
Comment (thewilsonator) [inline compiler/src/dmd/backend/dtype.d:1195]: did you mean to remove this?

File: compiler/src/dmd/backend/gother.d [hunk 2]
```diff
@@ -78,6 +77,7 @@ nothrow:
  *      Elemdata entry if found,
  *      null if not
  */
+
 @trusted
```
Comment (thewilsonator) [inline compiler/src/dmd/backend/gother.d:81]: ditto

**Review by thewilsonator — APPROVED** [review]
Looks good apart from unrelated whitespace changes near `@trusted`. If you did mean to remove those, please do so. If you didn't, please remove the whitespace changes.

File: compiler/src/dmd/backend/dtype.d (hunk 1 above)
Comment (dkorpel) [reply compiler/src/dmd/backend/dtype.d]: I removed it, found out it didn't work when compiling without dip1000, and restored it, but forgot the extra whitespace.

**Pull request merged**